### PR TITLE
build: apply a typical project boilerplate of BigScience repo

### PR DESCRIPTION
**Pull request by tianjianjiang**
Prepare GitHub actions for more regression tests in the future.

A side note: Poetry is optional for managing venv and dependencies, but syncing requirements(-dev).txt must be done manually for now.

I will describe some details with code review comments.

### CODE REVIEW

File: evaluation/tasks/tydiqa_primary/tydiqa_primary.py [hunk 1]
```diff
@@ -16,30 +17,31 @@
     {{"\n"}}{{context}}
     {%- endif -%}
     {{"\n"}}Answer: 
-    """
+    """  # noqa W291
```
Comment (tianjianjiang) [inline evaluation/tasks/tydiqa_primary/tydiqa_primary.py:20]: Although the trailing whitespace after `Answer:` is understandable, `load_dataset()` may still remove it with tokenizer. See also the test.

File: evaluation/tasks/tydiqa_secondary/tydiqa_secondary.py [hunk 2]
```diff
@@ -21,7 +20,7 @@
     {{"\n"}}{{context}}
     {%- endif -%}
     {{"\n"}}Answer: 
-    """
+    """  # noqa W291
```
Comment (tianjianjiang) [inline evaluation/tasks/tydiqa_secondary/tydiqa_secondary.py:23]: Same as https://github.com/bigscience-workshop/evaluation/pull/56/files#r690593718

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+[tool.black]
+line-length = 119
+target-version = ['py38']
+
+[tool.pytest.ini_options]
+filterwarnings = ["ignore:the imp module is deprecated in favour of importlib:DeprecationWarning:tensorflow.*"]
```
Comment (tianjianjiang) [inline pyproject.toml:6]: https://github.com/tensorflow/tensorflow/issues/31412

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+[tool.poetry]
+name = "evaluation"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
+
+[tool.poetry.dependencies]
+python = "^3.8.11"
```
Comment (tianjianjiang) [inline pyproject.toml:15]: `python>=3.8.10` supports Apple Silicon and macOS BigSur.
Personally I'm not using them. Just to ensure no one will got surprised by an incompatible version.

File: tests/test_tydiqa_primary.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    tokenizer.pad_token = tokenizer.eos_token
+    tokenizer.padding_side = "left"
+    dataset = TyDiQADataset(data, tokenizer, ["english"])
+    prompt = dataset[0]["prompt"]
+    assert (
+        "Wound care encourages and speeds wound healing via cleaning and protection from reinjury or infection. "
+        "Depending on each patient's needs, it can range from the simplest first aid to entire nursing specialties "
+        "such as wound, ostomy, and continence nursing and burn center care.\n"
+    ) in prompt
+    assert prompt.endswith("Answer:")
```
Comment (tianjianjiang) [inline tests/test_tydiqa_primary.py:19]: The trailing whitespace is gone, cf. https://github.com/bigscience-workshop/evaluation/pull/56/files#r690593718.

File: tests/test_tydiqa_primary.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    tokenizer = AutoTokenizer.from_pretrained("gpt2")
+    tokenizer.pad_token = tokenizer.eos_token
+    tokenizer.padding_side = "left"
+    dataset = TyDiQADataset(data, tokenizer, ["english"])
+    prompt = dataset[0]["prompt"]
+    assert (
+        "Wound care encourages and speeds wound healing via cleaning and protection from reinjury or infection. "
+        "Depending on each patient's needs, it can range from the simplest first aid to entire nursing specialties "
+        "such as wound, ostomy, and continence nursing and burn center care.\n"
+    ) in prompt
```
Comment (tianjianjiang) [inline tests/test_tydiqa_primary.py:18]: This is just a demo, not really a good test case. For better test cases, perhaps more PRs wanted.

File: tests/test_tydiqa_secondary.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    tokenizer = AutoTokenizer.from_pretrained("gpt2")
+    tokenizer.pad_token = tokenizer.eos_token
+    tokenizer.padding_side = "left"
+    dataset = TyDiQADataset(tokenizer, ["english"])
+    prompt = next(iter(dataset))["prompt"]
+    assert (
+        "Wound care encourages and speeds wound healing via cleaning and protection from reinjury or infection. "
+        "Depending on each patient's needs, it can range from the simplest first aid to entire nursing specialties "
+        "such as wound, ostomy, and continence nursing and burn center care.\n"
+    ) in prompt
```
Comment (tianjianjiang) [inline tests/test_tydiqa_secondary.py:17]: This is just a demo, not really a good test case. For better test cases, perhaps more PRs wanted.

File: tests/test_tydiqa_secondary.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    assert prompt.endswith("Answer:")
```
Comment (tianjianjiang) [inline tests/test_tydiqa_secondary.py:18]: The trailing whitespace is gone, cf. https://github.com/bigscience-workshop/evaluation/pull/56/files#r690593718.

**Comment by jaketae** [comment]
Hey @tianjianjiang, thanks for the PR. The changes you suggest will definitely help maintain this repo in a more organized, systematic fashion. I like the simple `Makefile` as well as the suggestion to use GitHub workflows for general formatting, style checks, and unit testing.

However, I think merging this will cause a host of issues in other PRs that are also in the line. Would it be possible to first merge only newly added files and not the whitespace/styling edits by `black` and `isort`? Or if you have any other suggestions on how to proceed logistically, that would also be helpful. Thanks!

**Review by jaketae — APPROVED** [review]
Thanks for the PR. I generally agree with the line of changes you suggest, i.e. unit testing, GitHub workflows, and enforcing code style. Contrary to my original thought, the PR does not involve many merge conflicts (yet). Let's go ahead with the changes for now and brush up on the details as we go. Thanks again!

**Comment by tianjianjiang** [comment]
> Contrary to my original thought, the PR does not involve many merge conflicts (yet). Let's go ahead with the changes for now and brush up on the details as we go. LGTM!

Sorry for the late reaction. I could have left those style changes out but too late for that. Thank you for the understanding!

**Pull request merged**